jupyterhub/templates/proxy/deployment.yaml

```diff
@@ -24,9 +24,9 @@ spec:
         {{- . | toYaml | trimSuffix "\n" | nindent 8 }}
         {{- end }}
       annotations:
-        # This restarts the pod when the proxy secret changes.
-        # The proxy secret is *only* present when doing manual HTTPS
-        # certs, so restarting is required to get those changes.
+        # We want to restart proxy only if the auth token changes
+        # Other changes to the hub config should not restart
+        checksum/auth-token: {{ include "jupyterhub.hub.config.ConfigurableHTTPProxy.auth_token" . | sha256sum }}
```

Comment:
If someone has a random 8 character long token, the sha256sum could be recognized by a hash dictionary - that is why I wanted to trunc it to 4 characters (65k permutations still).

Reply:
I don't see this as a risk for the provided length cert, key and misc YAML hashed in the proxy-secret checksum though.

Reply:
@consideRatio hmm, can you tell me what you mean by 'recognized by a hash dictionary'? I'm slightly confused :)

Reply:
By using brute force you can hash all combinations of for characters up to X characters. Then, my concern is that such has would be recognizable as the hash within the pod spec and be used to lookup the actual auth_token.

Reply:
Assuming the password was less than X in length, so not 32 characters or such but for example 10

Reply:
@consideRatio aaaaaaaaaaaaaah, that's very cool and clever - thank you. I'll add the trunc with an explanation.